tests/api.test.js

```
@@ -28,5 +28,208 @@ describe('API tests', () => {
+            request(app)
+                .post('/rides')
+                .type("form")
+                .send({
+                    'start_lat': -95,
+                    'start_long': 95,
+                    'end_lat': -190,
+                    'end_long': 190,
+                    'rider_name': 1,
+                    'driver_name': 2,
+                    'driver_vehicle': 3 
+                })
+                .expect('Content-Type', 'application/json; charset=utf-8')
+                .expect(400, done);
+        });
+    });
+
+    describe('GET /rides', () => {
+        it('should return list of all rides (200)', (done) => {
+            request(app)
+                .get('/rides')
+                .expect('Content-Type', 'application/json; charset=utf-8')
+                .expect(200, done);
+        });
+
+        it('should return list of all rides in page 1 with limit 10', (done) => {
+            request(app)
+                .get('/rides?page=1&limit=10')
+                .expect('Content-Type', 'application/json; charset=utf-8')
+                .expect(200, done);
+        });
+
+        it('should return list of all rides without limit due to only able to limit if page and limit given', (done) => {
+            request(app)
+                .get('/rides?page=1')
+                .expect('Content-Type', 'application/json; charset=utf-8')
+                .expect(200, done);
+        });
+
+        it('should return list of all rides without limit due to only able to limit if page and limit given', (done) => {
+            request(app)
+                .get('/rides?limit=10')
+                .expect('Content-Type', 'application/json; charset=utf-8')
+                .expect(200, done);
+        });
+
+        it('should return list of all rides in page 1 with limit 10', (done) => {
+            request(app)
+                .get('/rides?page=search&limit=search')
```

one would expect a query like this to return 400. Not returning an error here is considered to be a silent error on the client side, this will cause confusion on the client side